test/e2e/framework/expect.go

```diff
@@ -43,6 +47,18 @@ func ExpectNoError(err error, explain ...interface{}) {
 // ExpectNoErrorWithOffset checks if "err" is set, and if so, fails assertion while logging the error at "offset" levels above its caller
 // (for example, for call chain f -> g -> ExpectNoErrorWithOffset(1, ...) error would be logged for "f").
 func ExpectNoErrorWithOffset(offset int, err error, explain ...interface{}) {
+	if err != nil &&
+		errors.Is(err, e2eerrors.ErrFailureMsg) {
```

Comment:
That works for me. I didn't want to go that far because I wasn't sure whether it would reduce usability in other cases, but logging the error inside the test output should prevent that.